tree/ntupleutil/v7/src/RNTupleInspector.cxx

```diff
@@ -0,0 +1,103 @@
+/// \file RNTupleInspector.cxx
+/// \ingroup NTuple ROOT7
+/// \author Florine de Geus <florine.willemijn.de.geus@cern.ch>
+/// \date 2023-01-09
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <ROOT/RError.hxx>
+#include <ROOT/RNTuple.hxx>
+#include <ROOT/RNTupleDescriptor.hxx>
+#include <ROOT/RNTupleInspector.hxx>
+#include <ROOT/RError.hxx>
+
+#include <cstring>
+
+void ROOT::Experimental::RNTupleInspector::CollectSizeData()
+{
+   int compressionSettings = -1;
+   std::uint64_t compressedSize = 0;
+   std::uint64_t uncompressedSize = 0;
+
+   for (const auto &clusterDescriptor : fSourceNTupleDescriptor->GetClusterIterable()) {
+      compressedSize += clusterDescriptor.GetBytesOnStorage();
+
+      if (compressionSettings == -1) {
+         compressionSettings = clusterDescriptor.GetColumnRange(0).fCompressionSettings;
+      }
+   }
+
+   for (uint64_t colId = 0; colId < fSourceNTupleDescriptor->GetNColumns(); ++colId) {
+      const ROOT::Experimental::RColumnDescriptor &colDescriptor = fSourceNTupleDescriptor->GetColumnDescriptor(colId);
+
+      uint64_t elemSize =
+         ROOT::Experimental::Detail::RColumnElementBase::Generate(colDescriptor.GetModel().GetType())->GetSize();
+
+      uint64_t nElems = fSourceNTupleDescriptor->GetNElements(colId);
+      uncompressedSize += nElems * elemSize;
+   }
+
+   fCompressionSettings = compressionSettings;
+   fCompressedSize = compressedSize;
+   fUncompressedSize = uncompressedSize;
+}
+
+ROOT::Experimental::RResult<std::unique_ptr<ROOT::Experimental::RNTupleInspector>>
+ROOT::Experimental::RNTupleInspector::Create(std::unique_ptr<RNTupleReader> &sourceNTupleReader)
```

Comment:
Since we really only need the descriptor, I think we should not rely on the `RNTupleReader` at all. We can have a factory method from the page source or from `RNTuple *`, which can generate a page source.